Sources/TwitchIRC/IncomingMessage/PrivateMessage.swift

```diff
@@ -143,10 +168,64 @@ public struct PrivateMessage: MessageWithBadges {
             groupsOfExcludedUnavailableKeys: occasionalKeys
         )
     }
+
+    private func parseEmotes(from emoteString: String) -> [EmoteReference] {
+        /// First, we need to make sure we have an homogeneous array. In the message, emotes come as
+        /// a string that looks like `<id>:<start>-<end>/<id>:<start>-<end>`. However, when the same
+        /// emote repeats multiple times, it's more like
+        /// `<id>:<start>-<end>,<start>-<end>,<start>-<end>/<id>:<start>-<end>` and the parser has
+        /// issues giving it back as a list (the parser uses commas to sepparate, and doesn't take
+        /// the `/` into account).
+        ///
+        /// To solve this, we ask the parser to read it as a string and we treat the whole thing.
+        ///
+        /// Splitting by `/` returns an array like
+        /// `["<id>:<start>-<end>", "<start>-<end>", "<start>-<end>", "<id>:<start>-<end>"]`,
+        /// which makes it hard to treat all the elements the same since some of them have no `<id>`.
+        /// We solve this by using the last seen id as the id for elements that don't have one.
+        ///
+        /// Ater that, it's just a matter of parsing the indices and finding the equivalent text in
+        /// the chat message. With those, we can create our `EmoteReference` instance and save it.
+        var parsed: [EmoteReference] = []
+
+        let emoteArray = emoteString.split(separator: "/")
+            .flatMap { $0.split(separator: ",") }
+            .map { String($0) }
+
+        var lastID: String? = nil
+        for emote in emoteArray {
+            /// This gives us either `["<id>", "<start>", "<end>"]` or `["<start>", "<end>"]`
+            var parts = emote.split(separator: ":")
+                .flatMap { $0.split(separator: "-") }
+                .map { String($0) }
+
+            /// If we have an id, save it
+            if parts.count == 3 {
+                lastID = parts.removeFirst()
+            }
+
+            /// Try and retrieve the parts and add them to our list of emotes
+            if let lastID,
```

Comment:
We cannot use this kind of variable binding because we need to support older swift versions than 5.7.
change this to `let lastID = lastID,`

Reply:
Got it! That's my bad, I could only run the tests on Xcode 14 :)